app/src/main/java/sdp/moneyrun/ui/game/GameLobbyActivity.java

```diff
@@ -3,43 +3,88 @@
 import android.content.Intent;
 import android.os.Bundle;
 import android.util.Log;
-import android.view.View;
-import android.widget.Button;
+import android.widget.ListView;
 import android.widget.TextView;
 
 import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
 
-import com.google.android.gms.tasks.OnCompleteListener;
-import com.google.android.gms.tasks.Task;
 import com.google.firebase.database.DataSnapshot;
 import com.google.firebase.database.DatabaseError;
 import com.google.firebase.database.GenericTypeIndicator;
 import com.google.firebase.database.ValueEventListener;
 
+import java.util.ArrayList;
 import java.util.List;
 
 import sdp.moneyrun.R;
+import sdp.moneyrun.database.DatabaseProxy;
 import sdp.moneyrun.database.GameDatabaseProxy;
-import sdp.moneyrun.database.PlayerDatabaseProxy;
 import sdp.moneyrun.game.Game;
 import sdp.moneyrun.player.Player;
 import sdp.moneyrun.ui.menu.MenuActivity;
 
+
 public class GameLobbyActivity extends AppCompatActivity {
     private Game game;
     private String gameId;
     private Player user;
-
+    private ArrayList<Player> playerList = new ArrayList<>();
+    private LobbyPlayerListAdapter listAdapter;
+    TextView playerMissingView;
+    DatabaseProxy dbProxy;
+    private int playerInGame;
+    private int missingPlayers;
+    private ListView playerListView;
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_game_lobby);
+        addQuitLobbyButton();
+        addAdapter();
+        playerInGame = 1;
+        playerMissingView = findViewById(R.id.lobby_players_missing_TextView);
+        String default_player_missing = getString(R.string.lobby_player_missing, missingPlayers);
+        playerMissingView.setText(default_player_missing);
         gameId = (String) getIntent().getStringExtra (getResources().getString(R.string.join_game_lobby_intent_extra_id));
         user = (Player) getIntent().getSerializableExtra(getResources().getString(R.string.join_game_lobby_intent_extra_user));
         runFunctionalities();
+
+    }
+
+
+    public void addQuitLobbyButton(){

```

Comment:
Isn't there already a leave lobby button? And when you go back to the Menu you need to pass the player as an extra.

Reply:
There is one adter the addQuitLobbyButton function call not before?